x-pack/plugin/eql/src/main/java/org/elasticsearch/xpack/eql/analysis/Analyzer.java

```diff
@@ -51,6 +61,60 @@ private LogicalPlan verify(LogicalPlan plan) {
         return plan;
     }
 
+    private static class ReplaceWildcards extends AnalyzeRule<Filter> {
+
+        private static boolean isWildcard(Expression expr) {
+            if (expr instanceof Literal) {
+                Literal l = (Literal) expr;
+                if (l.value() instanceof String) {
+                    String s = (String) l.value();
+                    return s.contains("*");
```

Comment:
Would be possible to add more tests that also look at scenarios involving escape characters and [all types of string that eql supports](https://eql.readthedocs.io/en/latest/query-guide/basic-syntax.html#strings)?
Can the `*` be escaped? If so, we should have a test covering this case.

Reply:
yeah, I can add more tests. And there's no escape for `*`. But if you want to perform an exact (+/- case sensitivity) comparison, you can put the wildcard on the left (https://github.com/elastic/elasticsearch/pull/51901#discussion_r375360920). Whether this functionality is good or not is a fair question, and I think it's fair to change this because I doubt any users are aware of the workaround.
```
field == "*wildcard*" <==> field LIKE "%wildcard%"
"*wildcard*" == field <==> field EQ "%wildcard%"
```

Reply:
To be honest, I find this more like a bug, rather than feature (that implies an workaround).
A == B and B == A should be equivalent imo. Equality is not a predicate (like `LIKE`) where a certain element sits on the right and another one sits on the left and they are not interchangeable.